templates/utils.js

```diff
@@ -10,8 +10,12 @@ const replaceByClonedSource = options => {
 };
 
 const deepMergeWithoutKeysOrder = createDeepMerge({ mergeArray: replaceByClonedSource });
+const deepMergeArray = createDeepMerge();
 
 export const deepMerge = (...args) => {
+  if (Array.isArray(args[0])) {
+    return deepMergeArray(...args);
+  }
```

Comment:
Added this extra branch, so that while working with arrays example in `Header.tsx.template.mjs` (extraMenuLinksObjects) we concat them instead of the below object merge logic